MinecraftClient/Program.cs

```diff
@@ -229,12 +229,27 @@ private static void InitializeClient()
                 if (Settings.DebugMessages)
                     Translations.WriteLine("debug.session_id", session.ID);
 
-                //ProtocolHandler.RealmsListWorlds(Settings.Username, PlayerID, sessionID); //TODO REMOVE
+                ProtocolHandler.RealmsListWorlds(Settings.Username, session.PlayerID, session.ID);
 
                 if (Settings.ServerIP == "")
                 {
                     Translations.Write("mcc.ip");
-                    Settings.SetServerIP(Console.ReadLine());
+                    string addressInput = Console.ReadLine();
+                    if (addressInput.StartsWith("realms:"))
+                    {
+                        string worldId = addressInput.Substring(7, addressInput.Length - 7);
+                        string RealmsAddress = ProtocolHandler.GetRealmsWorldServerAddress(worldId, Settings.Username, session.PlayerID, session.ID);
+                        if (RealmsAddress != "")
+                        {
+                            addressInput = RealmsAddress;
+                        }
+                        else
+                        {
+                            // TODO: Handle failure
+                            Exit(0);
+                        }
+                    }
+                    Settings.SetServerIP(addressInput);
                 }
 
                 //Get server version

```

Comment:
Are Realms world always vanilla and of the most recent version? In that case we can skip all this part as @LesterLian says [ping does not work](https://github.com/ORelio/Minecraft-Console-Client/pull/1533#issuecomment-817165765), and use MCHighestVersion and forge = null here.

Reply:
I think that's true. Even some datapacks don't work on Realms. Also, I just uploaded a 1.15.2 world and it has been converted into 1.16.5. Skip the next two blocks seems to work just fine, but I need someone else to make sure I really join the server.